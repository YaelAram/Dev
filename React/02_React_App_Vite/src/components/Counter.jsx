import React, { useReducer } from 'react';
import PropTypes from 'prop-types';

const handdleCounter = ( currentValue, { type, init } ) => {
    switch( type ){
        case 'add': return currentValue + 1;
        case  'substract': return currentValue - 1;
        default: return init;
    }
};

function Counter( { value } ){
    const [ counter, dispath ] = useReducer( handdleCounter, value );

    return (
        <React.Fragment>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>
            <input type="button" aria-label="btn-substract" value="Substract 1" onClick={ () => dispath( { type: 'substract' } ) }/>
            <input type="button" aria-label="btn-reset" value="Reset" onClick={ () => dispath( { type: 'reset', init: value } ) }/>
            <input type="button" aria-label="btn-add" value="Add 1" onClick={ () => dispath( { type: 'add' } ) }/>
        </React.Fragment>
    );
};

Counter.propTypes = {
    value: PropTypes.number.isRequired
};

export default Counter;
