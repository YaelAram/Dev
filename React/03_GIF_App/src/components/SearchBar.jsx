import React, { useRef } from 'react';
import PropTypes from 'prop-types';

export function SearchBar( { onAddTopic } ) {
    const inputSearch = useRef( '' );

    const handleSubmit = ( event ) => {
        event.preventDefault();
        const newTopic = inputSearch.current.value.trim().toUpperCase();
        if( !newTopic ) return;
        onAddTopic( newTopic );
        inputSearch.current.value = '';
    };

    return (
        <form onSubmit={ handleSubmit } aria-label="form">
            <input type="text" placeholder="e.g. Dr. House" ref={ inputSearch } autoFocus />
        </form>
    );
};

SearchBar.propTypes = {
    onAddTopic: PropTypes.func.isRequired
};
