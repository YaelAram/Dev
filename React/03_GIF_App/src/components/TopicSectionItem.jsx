import React from 'react';
import PropTypes from 'prop-types';

export function TopicSectionItem( { title, url } ) {
    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    );
};

TopicSectionItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};
