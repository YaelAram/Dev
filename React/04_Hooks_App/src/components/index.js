export * from './BreakingBad';
export * from './Counter';
export * from './Counter2';
export * from './Counter3';
export * from './Counter4';
export * from './CounterLabel';
export * from './Child';
export * from './IncrementButton';
export * from './LoadingAlert';
export * from './Memorize';
export * from './Message';
export * from './Father';
export * from './Quote';
export * from './SimpleForm';
export * from './SimpleForm2';
