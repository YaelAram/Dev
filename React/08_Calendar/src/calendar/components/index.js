export * from './CalendarBox';
export * from './CalendarModal';
export * from './FabDeleteEvent';
export * from './FabNewEvent';
export * from './NavBar';
