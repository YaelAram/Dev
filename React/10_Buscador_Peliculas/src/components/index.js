export * from "./ListOfMovies";
export * from "./Loader";
export * from "./Movies";
