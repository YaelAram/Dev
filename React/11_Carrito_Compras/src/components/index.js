export * from "./Cart";
export * from "./Filters";
export * from "./Header";
export * from "./Icons";
export * from "./Products";
