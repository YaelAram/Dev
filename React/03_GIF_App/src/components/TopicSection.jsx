import React from 'react';
import PropTypes from 'prop-types';

import { useFetch } from '../hooks/useFetch';
import { TopicSectionItem } from './TopicSectionItem';

export function TopicSection( { topic, onDeleteTopic } ) {
    const images = useFetch( topic );

    const handleOnClickDelete = () => {
        onDeleteTopic( topic );
    };

    return (
        <section>
            <h2>{ topic }</h2>
            <input type="button" value="Delete Topic" onClick={ handleOnClickDelete } />
            <div className="card-grid">
                { images.map( ( image ) => <TopicSectionItem key={ image.id } { ...image } /> ) }
            </div>
        </section>
    );
};

TopicSection.propTypes = {
    topic: PropTypes.string.isRequired,
    onDeleteTopic: PropTypes.func.isRequired
}
