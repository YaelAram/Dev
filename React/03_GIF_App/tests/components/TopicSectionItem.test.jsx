import { screen, render, fireEvent } from '@testing-library/react';
import { TopicSectionItem } from '../../src/components/TopicSectionItem';

describe( 'TopicSectionItem component', () => {
    const topic = { title: 'Rammstein', url: 'https://rammstein.com/rammstein.gif' };

    test( 'Snapshot TopicSectionItem component', () => {
        const { container } = render( <TopicSectionItem { ...topic } /> );
        expect( container ).toMatchSnapshot();
    } );

    test( 'Check image container', () => {
        render( <TopicSectionItem { ...topic } /> );
        const { alt, src } = screen.getByRole( 'img' );
        expect( alt ).toBe( topic.title ); 
        expect( src ).toBe( topic.url );
    } );

    test( 'Check image container', () => {
        render( <TopicSectionItem { ...topic } /> );
        // It does not match the alt="title" because it is a property
        expect( screen.getByText( topic.title ) ).toBeTruthy();
    } );
} );
