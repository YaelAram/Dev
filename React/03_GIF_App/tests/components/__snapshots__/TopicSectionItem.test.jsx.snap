// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicSectionItem component Snapshot TopicSectionItem component 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Rammstein"
      src="https://rammstein.com/rammstein.gif"
    />
    <p>
      Rammstein
    </p>
  </div>
</div>
`;
